aleph-client/src/contract/event.rs

```diff
@@ -0,0 +1,130 @@
+//! Utilities for listening for contract events.
+//!
+//! To use the module you will need to pass a connection, some contracts and an `UnboundedSender` to the
+//! [listen_contract_events] function. You most likely want to `tokio::spawn` the resulting future, so that it runs
+//! concurrently.
+//!
+//! ```no_run
+//! # use std::sync::Arc;
+//! # use std::sync::mpsc::channel;
+//! # use std::time::Duration;
+//! # use aleph_client::{AccountId, Connection, SignedConnection};
+//! # use aleph_client::contract::ContractInstance;
+//! # use aleph_client::contract::event::{listen_contract_events};
+//! # use anyhow::Result;
+//! use futures::{channel::mpsc::unbounded, StreamExt};
+//!
+//! # async fn example(conn: Connection, signed_conn: SignedConnection, address1: AccountId, address2: AccountId, path1: &str, path2: &str) -> Result<()> {
+//! // The `Arc` makes it possible to pass a reference to the contract to another thread
+//! let contract1 = Arc::new(ContractInstance::new(address1, path1)?);
+//! let contract2 = Arc::new(ContractInstance::new(address2, path2)?);
+//!
+//! let conn_copy = conn.clone();
+//! let contract1_copy = contract1.clone();
+//! let contract2_copy = contract2.clone();
+//!
+//! let (tx, mut rx) = unbounded();
+//! let listen = || async move {
+//!     listen_contract_events(&conn, &[contract1_copy.as_ref(), contract2_copy.as_ref()], tx).await?;
+//!     <Result<(), anyhow::Error>>::Ok(())
+//! };
+//! let join = tokio::spawn(listen());
+//!
+//! contract1.contract_exec0(&signed_conn, "some_method").await?;
+//! contract2.contract_exec0(&signed_conn, "some_other_method").await?;
+//!
+//! println!("Received event {:?}", rx.next().await);
+//!
+//! rx.close();
+//! join.await??;
+//!
+//! #   Ok(())
+//! # }
+//! ```
+
+use std::collections::HashMap;
+
+use anyhow::{bail, Result};
+use contract_transcode::Value;
+use futures::{channel::mpsc::UnboundedSender, StreamExt};
+
+use crate::{contract::ContractInstance, AccountId, Connection};
+
+/// Represents a single event emitted by a contract.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct ContractEvent {
+    /// The address of the contract that emitted the event.
+    pub contract: AccountId,
+    /// The name of the event.
+    pub ident: Option<String>,
+    /// Data contained in the event.
```

Comment:
Maybe just "event data"?

Reply:
If you're suggesting that a field named `data` in a struct named `Event` be commented `event data`, then I strongly disagree. This does not add any information and doesn't even put it in a more "human readable" form. I realize that the comment as-is doesn't add any information either, but at least it's a nominal sentence.